### PR TITLE
(Quick and dirty) automate documentation

### PR DESCRIPTION
**Pull request by gcanti**
@raveclassic this is another attempt to generate documentation from source files (I developed a custom script).

### CODE REVIEW

File: docs/api/parser.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+const parseTypeAliasDeclarationData = (sf: SourceFile, tad: TypeAliasDeclaration): Option<Data> => {
+  const annotation = getAnnotation(tad.getDocumentationCommentNodes())
+  if (isData(annotation)) {
+    const dataName = tad.getName()
+    const signature = tad.getText().substring('export '.length)
+    const description = fromJSDocDescription(annotation.description)
+    const constructors: Array<Constructor> = annotation.tags.filter(tag => isConstructor(tag)).map(tag => {
+      const name = tag.name
+      if (typeof name === 'undefined') {
+        throw new Error(`Missing constructor name in ${dataName}`)
```
Comment (raveclassic) [inline docs/api/parser.ts]: Should pure fp library use `throw` in it's sources? Isn't it a poor example? ;)

Comment (gcanti) [reply docs/api/parser.ts]: eh... I added "Quick and dirty" in the title specifically for that 😛 (and some poor string parsing..)

More seriously, if the result feels ok we can improve the code at a later stage

**Comment by gcanti** [comment]
(ominous throws removed)

@raveclassic please take a look at the general result here and let me know what you think

[API index](https://github.com/gcanti/fp-ts/blob/776aa1d7162261e0fde19fbc6a2c871abcda03ae/docs/api/md/index.md)

The relevant info should be there (and we can change the template anytime)

File: docs/api/parser.ts (hunk 1 above)
Comment (raveclassic) [reply docs/api/parser.ts]: Sounds good :)

**Comment by raveclassic** [comment]
Everything looks fine

**Comment by gcanti** [comment]
@raveclassic thanks for reviewing 👍

**Pull request merged**